static/js/buddy_data.js

```diff
@@ -247,30 +247,79 @@ function maybe_shrink_list(user_ids, filter_text) {
 
 exports.get_filtered_and_sorted_user_ids = function (filter_text) {
     var user_ids;
-
-    if (filter_text) {
-        // If there's a filter, select from all users, not just those
-        // recently active.
-        user_ids = filter_user_ids(filter_text, people.get_active_user_ids());
-    } else {
-        // From large realms, the user_ids in presence may exclude
-        // users who have been idle more than three weeks.  When the
-        // filter text is blank, we show only those recently active users.
-        user_ids = presence.get_user_ids();
+    var use_member_list = page_params.user_list_content === "member";
+    var use_pm_with_filter = false;
+    var use_stream_filter = false;
+    var filter;
+    if (use_member_list) {
+        var active = narrow_state.active();
+        filter = narrow_state.filter();
```

Comment:
We have a serious problem with overuse of the term "filter" here making it hard to understand.  

I think probably some good names would be: `narrow_filter` if you want to interact with that object, and renaming `filter_text` to `user_filter_text`.  I would do at least that second rename as a prep commit to make the rest more readable.

Reply:
I also think I should switch over `user_member_list` to `should_use_member_list`, `use_pm_with_filter` to `should_use_pm_member_list` and `should_use_stream_member_list`. `filter` to `narrow_filter` and `filter_text` to `user_filter_text`.